### DISCUSSION TITLE
directus_users has a definition that does not allow to re-import data from SQLite dump

### DISCUSSION THREAD

**Discussion by Pachat** [directus/directus#12848] [Archive]
### Describe the Bug

Dump from SQLite are not re-importable due to a CREATE definition with types unknown to SQLite.

### To Reproduce

- Dump an SQLite Directus DB 
   or run
   `select sql from  sqlite_master  where name = 'directus_users';`

### Errors Shown

The result from a new installation of 9.9.0 is

```sql
CREATE TABLE "directus_users" (
`id` char(36) NOT NULL, 
`first_name` varchar(50), 
`last_name` varchar(50), 
`email` varchar(128), 
`password` varchar(255), 
`location` varchar(255), 
`title` varchar(50), 
`description` text, 
`tags` json, 
`avatar` char(36), 
`language` varchar(255) DEFAULT null, 
`theme` varchar(20) DEFAULT 'auto', 
`tfa_secret` varchar(255), 
`status` varchar(16) NOT NULL DEFAULT 'active', 
`role` char(36), 
`token` varchar(255), 
`last_access` datetime, 
`last_page` varchar(255), 
`provider` varchar(128) NOT NULL DEFAULT 'default', 
`external_identifier` varchar(255), 
`auth_data` json, 
`email_notifications` boolean DEFAULT '1', 
PRIMARY KEY (`id`), 
FOREIGN KEY (`role`) REFERENCES `directus_roles` (`id`) ON DELETE SET NULL)
```

PROBLEMS:
1. SQLite has not **char** type
2. Neither **json** type

A CREATE TABLE that works is:

```sql
CREATE TABLE "directus_users" (
`id` varchar(36) NOT NULL, 
`first_name` varchar(50), 
`last_name` varchar(50), 
`email` varchar(128), 
`password` varchar(255), 
`location` varchar(255), 
`title` varchar(50), 
`description` text, 
`tags` text, 
`avatar` varchar(36), 
`language` varchar(255) DEFAULT null, 
`theme` varchar(20) DEFAULT 'auto', 
`tfa_secret` varchar(255), 
`status` varchar(16) NOT NULL DEFAULT 'active', 
`role` varchar(36), 
`token` varchar(255), 
`last_access` datetime, 
`last_page` varchar(255), 
`provider` varchar(128) NOT NULL DEFAULT 'default', 
`external_identifier` varchar(255), 
`auth_data` text, 
`email_notifications` boolean DEFAULT '1', 
PRIMARY KEY (`id`), 
FOREIGN KEY (`role`) REFERENCES `directus_roles` (`id`) ON DELETE SET NULL)
```
_Verified with WebStorm_

Same problem with the other tables that have a `char` or a `json`

### What version of Directus are you using?

9.9.0

### What version of Node.js are you using?

v16.14.0

### What database are you using?

SQLite 3.37

### What browser are you using?

Firefox

### How are you deploying Directus?

locally

**Comment by azrikahar** [comment]
Moving this to discussion as it sounds more of a SQLite database thing rather than a bug with Directus codebase itself.

With that said, would you mind elaborating on the `re-importable` statement? Do you mean using the same CREATE statement doesn't work? I did a test locally and it does work:

https://user-images.githubusercontent.com/42867097/163991191-0de231f6-bb14-4968-8728-16e1964baecf.mp4

**Reply by Pachat** [reply]
My try was done through the WebStorm SQLite console where, as I said, the first CREATE SQL posted  doesn't work.

Seeing your example, I gave another try but with the CLI. And it works correctly there too.
So, its more of a WebStorm limitation. I'll be careful at that for the future.
Thank you so much for your time.
